### PR TITLE
Added root and label class keys for autocomplete pickers

### PR DESCRIPTION
**Pull request by HHogg**
## Hey, I just made a Pull Request!

Currently theming/styling the autocomplete labels for scaffolder and the catalog isn't possible because they don't have anyway to target them specifically, so they have button styling which doesn't match the MUI form labels on other inputs. This PR adds some classes for them. 

Ideally, these components would follow a similar layout and structure with a wrapping FormControl and InputLabel, but that might introduce a breaking change and would be more work to refactor as not all these components have been built consistently (like some have wrapping labels, some have sibling labels). 

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))

### CODE REVIEW

**Review by mikeyhc — APPROVED** [review]

File: plugins/scaffolder-react/src/next/overridableComponents.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 The Backstage Authors
```
Comment (tudi2d) [inline plugins/scaffolder-react/src/next/overridableComponents.ts:2]:
```suggestion
 * Copyright 2024 The Backstage Authors
```

File: plugins/scaffolder-react/src/next/overridableComponents.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export type BackstageOverrides = Overrides & {
+  [Name in keyof ScaffolderReactComponentsNameToClassKey]?: Partial<
+    StyleRules<ScaffolderReactComponentsNameToClassKey[Name]>
+  >;
+};
+
+declare module '@backstage/theme' {
+  interface OverrideComponentNameToClassKeys
+    extends ScaffolderReactComponentsNameToClassKey {}
+}
```
Comment (tudi2d) [inline plugins/scaffolder-react/src/next/overridableComponents.ts:36]: not completely sure about this, as I do believe the `@alpha` export is correct here, but `declare module '@backstage/theme'` will extend the theme type even for non-alpha usage of the scaffolder, if this file is loaded. I can't think about a better solution though.

@benjdlambert is the `next` folder loaded at all, when people are using the legacy scaffolder?

**Review by tudi2d — APPROVED** [review]
`catalog` components look good - small comment on the `alpha` `scaffolder` exports.

File: plugins/scaffolder-react/src/next/overridableComponents.ts (hunk 2 above)
Comment (benjdlambert) [reply plugins/scaffolder-react/src/next/overridableComponents.ts:36]: We removed the legacy scaffolder exports now. Don't actually think that the `/alpha` endpoints contain much now though, aside from some `props` and some smaller components.

**Pull request merged**